src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -115,4 +113,33 @@ namespace winrt::Microsoft::Management::Deployment::implementation
         connectResult->Initialize(winrt::Microsoft::Management::Deployment::ConnectResultStatus::CatalogError, nullptr);
         return *connectResult;
     }
+
+    hstring PackageCatalogReference::AdditionalPackageCatalogArguments()
+    {
+        if (!IsComposite())
+        {
+            winrt::Microsoft::Management::Deployment::implementation::PackageCatalogInfo* catalogInfoImpl = get_self<winrt::Microsoft::Management::Deployment::implementation::PackageCatalogInfo>(m_info);
+            ::AppInstaller::Repository::SourceDetails sourceDetails = catalogInfoImpl->GetSourceDetails();
+            auto customHeader = catalogInfoImpl->GetSourceDetails().CustomHeader;
+            if (customHeader.has_value())
+            {
+                return winrt::to_hstring(customHeader.value());
+            }
+        }
+
+        return {};
+    }
+    void PackageCatalogReference::AdditionalPackageCatalogArguments(hstring const& value)
+    {
+        if (IsComposite())
+        {
+            // Can't set AdditionalPackageCatalogArguments on a composite. Callers should set it on each non-composite PackageCatalogReference in the composite.
+            throw winrt::hresult_error(ERROR_INVALID_STATE);

```

Comment:
Does this automatically convert a Win32 error to an HRESULT?  It doesn't look like it from the docs.  Also, it looks like you should use either `winrt::hresult_illegal_state_change` or `winrt::throw_hresult(E_ILLEGAL_STATE_CHANGE)` given that they have those defined over this Win32 error we thought of off the cuff.